stratagem-runner/src/stratagem_warning.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+// Usage: stratagem_warning [FS-NAME|FS-MOUNT-POINT] [NID1] [NID2] ...
+
+use libc;
+use std::env;
+use std::ffi::CStr;
+use std::io;
+use std::process;
+use stratagem_runner::error::StratagemError;
+
+#[derive(Debug, serde::Serialize)]
+#[serde(rename_all = "SCREAMING-KEBAB-CASE")]
+struct Record<'a> {
+    path: &'a str,
+    user: &'a str,
+    uid: u32,
+    gid: u32,
+    atime: i64,
+    mtime: i64,
+    ctime: i64,
+    fid: &'a str,
+}
+
+fn write_records(
+    device: &String,
+    args: impl IntoIterator<Item = String>,
+) -> Result<(), StratagemError> {
+    let out = io::stdout();
+    let mntpt = liblustreapi::search_rootpath(&device).map_err(|e| {
+        eprintln!("Failed to find rootpath({}) -> {:?}", device, e);
+        e
+    })?;
+    let mut wtr = csv::Writer::from_writer(out);
+
+    for fid in args {
+        if let Ok(path) = liblustreapi::fid2path(&device, &fid) {
+            let pb: std::path::PathBuf = [&mntpt, &path].iter().collect();
+            let fullpath = pb.to_str().unwrap();
+            let stat = liblustreapi::mdc_stat(&fullpath).map_err(|e| {
```

Comment:
Since `fullpath` gets converted to a `PathBuf`within this fn, you could just have this fn take a `PathBuf`. Then you won't need to do the potentially fallible conversion to `str`.